packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts

```diff
@@ -139,143 +177,31 @@ it.skip('Create a directly funded channel between two engines ', async () => {
     fundingStrategy: 'Direct',
     challengeDuration: ONE_DAY,
   };
-
-  // the type assertion is due to
-  // https://github.com/devanshj/rxjs-from-emitter/blob/master/docs/solving-some-from-event-flaws.md#the-way-fromevent-checks-if-the-first-argument-passed-is-an-emitter-or-not-is-incorrect
-  const postFundAPromise = fromEvent<SingleChannelOutput>(a as any, 'channelUpdated')
-    .pipe(take(2))
-    .toPromise();
-
-  const postFundBPromise = fromEvent<SingleChannelOutput>(b as any, 'channelUpdated')
-    .pipe(take(2))
-    .toPromise();
-
-  const channelClosedAPromise = fromEvent<SingleChannelOutput>(a as any, 'channelUpdated')
-    .pipe(take(4))
-    .toPromise();
-
-  //        A <> B
-  // PreFund0
-  const preFundA = await a.createChannel(channelParams);
-
-  const channelId = preFundA.channelResult.channelId;
-
   const aBalanceInit = await getBalance(aAddress);
   const bBalanceInit = await getBalance(bAddress);
+  const assetHolderBalanceInit = await getBalance(ethAssetHolderAddress);
 
-  expect(preFundA.channelResult).toMatchObject({
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  const resultB0 = await b.pushMessage(getPayloadFor(participantB.participantId, preFundA.outbox));
+  const response = await a.createChannels([channelParams]);
+  await waitForObjectiveProposals([response[0].objectiveId], b);
+  const bResponse = await b.approveObjectives([response[0].objectiveId]);
 
-  expect(getChannelResultFor(channelId, resultB0.channelResults)).toMatchObject({
-    status: 'proposed',
-    turnNum: 0,
-  });
+  await expect(response).toBeObjectiveDoneType('Success');
```

Comment:
Does this work consistently? If so, I am a bit surprised!

I could be misreading the code, but it seems to me that we are only waiting for the objective approval, not objective succeeding.

Reply:
This should be waiting for the objective to complete fully.

In the result from both `approveObjectives` and `createChannels` we return a `done` promise that resolves when the objective is completed.

The idea is that you get a result right away with a channelId/ObjectiveId  and a `promise`  that will resolve when the objective is completed.

Reply:
Got it! Thanks for the explanation!